### PR TITLE
Updating the existing CY14 backbone PR by Marco to work with 3.20.0 and recently refactored CY14 class

### PR DESCRIPTION
**Pull request by CB-quakemodel**
PR taking Marco's work implementing CY14 as modifiable by Boore et al. (2022) (https://github.com/gem/oq-engine/pull/8532) and updating to work in current version of the engine (CY14 was recently refactored a fair bit so easier to take the cybb branch and work on it in this PR, rather than resolve the merge conflicts and then work on cybb branch).

I have also added the gamma correction term to the distance scaling functions + a unit test (the path adjustment table I provide is currently for the central branch for SA(0.1 s) as provided in Table 2 of the paper).

The CY14 GMM always computes mean ground-motion for PGA as it can potentially be used for PSA with T < 0.3 if PSA(T) is less than PGA, therefore for the PGA coefficients in the path adjustment table I provide dummy values of 999. These dummy values are not used given the test is for T = 0.1 s (and this is asserted by checking the expected adjusted mean ground-motion is equal to the expected values for if adjusted using the T = 0.1s path adjustment coefficients).

Physically realistic values are obtained. For example, in the unit tests, if we use a target region stress drop of 120 bars and a host region stress drop of 100 bars, we observe an increase in the median from 0.066 g to 0.075 g. If we apply only the path adjustment we get a median of 0.046 g. If we apply the source scaling mentioned above and the path adjustment we get a median of 0.053 g. The rrup in the unit test ctx is about 55 km. Some sanity checks have also been performed e.g. if target and host region stress drops are the same with no path adjustment, identical median values of ground-shaking are observed.

### CODE REVIEW

File: openquake/hazardlib/gsim/chiou_youngs_2014.py [hunk 1]
```diff
@@ -470,35 +626,85 @@ class ChiouYoungs2014(GMPE):
 
     #: Reference shear wave velocity
     DEFINED_FOR_REFERENCE_VELOCITY = 1130
-
-    def __init__(self, sigma_mu_epsilon=0.0):
+        
+    def __init__(self, sigma_mu_epsilon=0.0, use_hw=True, add_delta_c1=False,
+                 alpha_nm=1.0, stress_par_host=None, stress_par_target=None,
+                 delta_gamma_tab=None):
+        super().__init__(sigma_mu_epsilon=sigma_mu_epsilon,
```
Comment (micheles) [inline openquake/hazardlib/gsim/chiou_youngs_2014.py:633]: The `super().__init__` is useless.

File: openquake/hazardlib/gsim/chiou_youngs_2014.py [hunk 2]
```diff
@@ -432,14 +578,34 @@ def get_mean_stddevs(name, C, ctx):
...
-    Implements GMPE developed by Brian S.-J. Chiou and Robert R. Youngs
+    Implements GMPE developed by Brian S.-J. Chiou and Robert R. Youngs.
 
-    Chiou, B. S.-J. and Youngs, R. R. (2014), "Updated of the Chiou and Youngs
+    Chiou, B. S.-J. and Youngs, R. R. (2014), "Update of the Chiou and Youngs
     NGA Model for the Average Horizontal Component of Peak Ground Motion and
     Response Spectra, Earthquake Spectra, 30(3), 1117 - 1153,
     DOI: 10.1193/072813EQS219M
-    """
-    adapted = False  # overridden in acme_2019
```
Comment (mmpagani) [inline openquake/hazardlib/gsim/chiou_youngs_2014.py:442]: Why are you removing this?

File: openquake/hazardlib/tests/gsim/data/C14adj/path_adjustment_table_target_region_idaho.txt
Comment (mmpagani) [inline openquake/hazardlib/tests/gsim/data/C14adj/path_adjustment_table_target_region_idaho.txt:1]: I would put this file into a folder called `chiou_youngs_2014` located in the same folder of the `chiou_youngs_2014.py` file

**Review by mmpagani — APPROVED** [review]
Apart from two minor comments, it looks good to me. Thanks Chris

File: openquake/hazardlib/tests/gsim/data/C14adj/path_adjustment_table_target_region_idaho.txt
Comment (CB-quakemodel) [reply openquake/hazardlib/tests/gsim/data/C14adj/path_adjustment_table_target_region_idaho.txt:1]: Moved as suggested

File: openquake/hazardlib/gsim/chiou_youngs_2014.py (hunk 2 above)
Comment (CB-quakemodel) [reply openquake/hazardlib/gsim/chiou_youngs_2014.py:442]: Readded

**Review by micheles — APPROVED** [review]

**Pull request merged**